src/main/java/io/vertx/core/impl/DeploymentManager.java

```diff
@@ -380,43 +383,94 @@ public void unregisterVerticleFactory(VerticleFactory factory) {
     return factoryList;
   }
 
+  private static URL mapToURL(String path) {
+    try {
+      File file = new File(path);
```

Comment:
Not sure if this works in all case. 

What about trying to build an url out of the string first and in case of failure try with File.

Reply:
I agree, what does the new file brings ?

Reply:
@vietj Nothing, I can also write new File(path).toURI().toURL();
@cescoffier Do you mean something like this?